### DISCUSSION TITLE
Icons based on entity state

### DISCUSSION THREAD

**Discussion by Rick100694** [Clooos/Bubble-Card#379] [Feature requests]
It would be nice if there is a opportunity to change the icons based on state of the entity.
I tried a couple of different codes but nothing does the work.

**Comment by Clooos** [comment]
Hi, this is in my plans. I just don't know when I will add that, I have a very big list. But this is in my priorities.

**Comment by neTix84** [comment]
Hi both,

for me worked the following code, i hope that helps...

`${icon.setAttribute("icon", state >=2 ? 'mdi:alert' : 'mdi:robot-vacuum')}`

or

`${icon.setAttribute("icon", hass.states['vacuum.downstairs'].state === 'error' ? 'mdi:alert' : 'mdi:robot-vacuum')}`
